extensions/amp-mustache/0.1/amp-mustache.js

```diff
@@ -0,0 +1,50 @@
+/**
+ * Copyright 2015 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {parse as mustacheParse, render as mustacheRender} from
+    '../../../thirdparty/mustache/mustache';
+import {childElementByTag} from '../../../src/dom';
+
+/** @const */
+const assert = AMP.assert;
+
+
+class AmpMustache extends AMP.BaseTemplate {
+
+  buildCallback() {
+    /** @private @const {string} */
+    this.template_ = this.element./*OK*/innerHTML;
+    mustacheParse(this.template_);
```

Comment:
I will rename this callback. It's actually what it does - lazy initialization. Poor naming.